src/Mod/Arch/ArchComponent.py

```diff
@@ -709,9 +709,10 @@ def onChanged(self,vobj,prop):
                             vobj.update()
         elif prop == "ShapeColor":
             # restore DiffuseColor after overridden by ShapeColor
-            if len(vobj.DiffuseColor) > 1:
-                d = vobj.DiffuseColor
-                vobj.DiffuseColor = d
+            if hasattr(vobj.Object,"DiffuseColor"):
```

Comment:
Error here: should be 
if hasattr(vobj,"DiffuseColor"):
and not
if hasattr(vobj.Object,"DiffuseColor"):